packages/lit-helpers/src/read-only-properties-mixin.js

```diff
@@ -0,0 +1,90 @@
+/** @typedef {import('lit-element').PropertyDeclaration & { readOnly: Boolean }} PropertyDeclarationWithReadOnly */
+
+/**
+ * @template TBase
+ * @typedef {new (...args: any[]) => TBase} Constructor
+ */
+
+/**
+ * Enables the nofity option for properties to fire change notification events
+ *
+ * @template TBase extends import('lit-element').LitElement
+ * @param {Constructor<TBase>} superclass
+ */
+export const ReadOnlyPropertiesMixin = superclass => {
+  /**
+   * @type {Map<string, symbol>}
+   * @private
+   */
+  const _readOnlyPropertyNamesMap = new Map();
+
+  // @ts-ignore
+  return class ReadOnlyPropertiesClass extends superclass {
+    /**
+     * @inheritdoc
+     * @param  {string} name property name
+     * @param  {PropertyDeclarationWithReadOnly} options augmented property declaration with optional `readOnly` boolean.
```

Comment:
```suggestion
     * @param  {PropertyDeclarationWithReadOnly | PropertyDeclaration} options augmented property declaration with optional `readOnly` boolean.
```

You will need to typedef the `PropertyDeclaration` above.